wrappers/android/examples/native_example/app/src/main/java/com/example/realsense_native_example/MainActivity.java

```diff
@@ -37,18 +44,25 @@ protected void onCreate(Bundle savedInstanceState) {
         //For multi activities applications use the application context instead of the activity context
         RsContext.init(getApplicationContext());
 
-        printMessage();
+        try {
+            // give T265 time to boot
+            sleep(1000);
+        } catch (InterruptedException e) {
+            Log.e(TAG, "onCreate: sleep interrupted");
+        }
```

Comment:
Is this needed given that you are using the `setDevicesChangedCallback` below?  After all it's the `RsContext` that boots the device?  Or at least it's the `rs2::context` in C++.  I'm not sure about Java.

Reply:
Application is unable to boot T265 upon start without explicitly putting main activity to sleep. Original code handles booting properly when camera is attached to the phone after app is already running, but when app tries to boot the camera on startup it goes into continuous loop trying to boot it every few milliseconds and crashes after several seconds

Reply:
RsContext starts the boot process which keeps getting interrupted and started over and over again, unless main activity is put to sleep. This code needs to run on UI thread, otherwise request for permission from Android's USB is not presented to the end user and app is not permitted to interact with T265 after it is rebooted and discovered with new vendor Id

Reply:
How about running `queryDevices()` after setting up `setDevicesChangedCallback` instead?  This will do the wait internal to `librealsense` and you can just throw away the result.

Reply:
tried that (there is call to [queryDevices()](https://github.com/smartynenko/librealsense/blob/e712bd5f28a97140d6439688192e84f33432e645/wrappers/android/examples/native_example/app/src/main/java/com/example/realsense_native_example/MainActivity.java#L98)) inside `onRSDeviceAttach` method. It will boot camera and call callback if app is already running when camera is attached, but will not if camera is connected [and not booted] before app starts.

I'd be happy to jump on #5492 wagon as soon as it joins the train ;)

Reply:
> tried that (there is call to queryDevices()) inside onRSDeviceAttach method.

Ah, I didn't see that.  Too bad.  I don't see why that wouldn't work. `queryDevices` should call `tm_boot` when it finds the boot loader and wait for the device to show up before returning.   Any ideas why it doesn't work?

Reply:
It does call `tm_boot`, but does not wait for it to finish booting (I see a lot of `Found a T265 to boot` messages in the log one after another). I could not figure out why this is happening

Reply:
I can't see why either.  It's supposed to work. :(

Reply:
Would you try removing the sleep while using this patch 7540a896bafbe84ed0ab812c447fd8b94479d53a?  There are too many sleeps already.

Reply:
looks like something changed in base code and all of my newly compiled tests are failing because of pipeline error. I will need to see if I am working of wrong bacaseline

A/libc: /usr/local/google/buildbot/src/android/ndk-release-r21/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:72: abort_message: assertion "terminating with uncaught exception of type rs2::error: Frame didn't arrive within 15000" failed
A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 30327 (_native_example)

Reply:
Giving another second to boot worked! No need to pause on startup :)

Reply:
Glad to hear!

Reply:
I ended up debugging the too short sleep when fixing something unrelated, and your description above led to realize that you were running into the exact same issue.